task_processing/interfaces/event.py

```diff
@@ -1,13 +1,18 @@
 from pyrsistent import field
+from pyrsistent import m
+from pyrsistent import PMap
 from pyrsistent import PRecord
 
-# TODO: organize and explain these
+from task_processing.interfaces.task_executor import TaskConfig
 
 
 class Event(PRecord):
     # reference to platform-specific event object
     raw = field()
     # is this the last event for a task?
     terminal = field(type=bool, mandatory=True)
+    success = field(type=(bool, type(None)), initial=None)
     # platform-specific event name
     platform_type = field(type=str)
+    task_config = field(type=TaskConfig)
+    middleware_data = field(type=PMap, initial=m())
```

Comment:
i don't think there is a better place for it actually. events are emitted from lowest level executor up to the upper layers and they get enriched with useful information along the way. we could say "why not just emit the pair of (event, mutable state) and let upper layers mutate the state rather than the event" but i would argue that it's already pretty much what we're doing with Event precord - only the `raw` is actual event, the rest is executor-provided data that helps upper layers (or end-user) to work with it.